File: 2551-apply-operations-to-an-array/apply-operations-to-an-array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        for indx in range(len(nums) - 1):
            if nums[indx] == nums[indx + 1]:
                nums[indx], nums[indx + 1] = 2* nums[indx], 0

        result = [0 for _ in range(len(nums))]

        indx = 0

        for num in nums:
            if num != 0:
                result[indx] = num
                indx += 1

        return result
        

File: 2551-apply-operations-to-an-array/apply-operations-to-an-array.go
func applyOperations(nums []int) []int {

    var result []int

    index := 0

    for index < len(nums) {
        if (index < len(nums) - 1) && nums[index] == nums[index + 1] {
            if nums[index] * 2 != 0{
                result = append(result, nums[index] * 2)
            }
            index += 2
        } else {
            if nums[index] != 0{
                result = append(result, nums[index])
            }
            index += 1
        }
    }

    for len(result) < len(nums){
        result = append(result, 0)
    }
    
    return result
}